### PR TITLE
Add performance test for filelog based kubernetes container logs receiver

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Add performance test for filelog based kubernetes container logs receiver

This PR contains one performance test (for kubernetes containerd logs with format autodetection)

**Link to tracking Issue:**
#2266

**Testing:**
Perf tests

**Documentation:**
N/A

### CODE REVIEW

File: testbed/datasenders/k8s.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+	"strconv"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/testbed/testbed"
+)
+
+
+// FileLogK8sWriter represenrts abstract container k8s writer
```
Comment (tigrannajaryan) [inline testbed/datasenders/k8s.go:31]:
```suggestion
// FileLogK8sWriter represents abstract container k8s writer
```

File: testbed/datasenders/k8s.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	return []byte(sb.String())
+}
+
+func (f *FileLogK8sWriter) Flush() {
+	_ = f.file.Sync()
+}
+
+func (f *FileLogK8sWriter) GenConfigYAMLStr() string {
+	// Note that this generates a receiver config for agent.
+	// We are testing stanza receiver here.
```
Comment (tigrannajaryan) [inline testbed/datasenders/k8s.go:127]:
```suggestion
	// We are testing filelog receiver here.
```

File: testbed/datasenders/k8s.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+	file   *os.File
+	config string
+}
+
+// Ensure FileLogK8sWriter implements LogDataSender.
+var _ testbed.LogDataSender = (*FileLogK8sWriter)(nil)
+
+// NewFileLogK8sWriter creates a new data sender that will write kubernetes containerd
+// log entries to a file, to be tailed by FileLogReceiver and sent to the collector.
+func NewFileLogK8sWriter(config string) *FileLogK8sWriter {
```
Comment (tigrannajaryan) [inline testbed/datasenders/k8s.go:43]: Add comment to explain what `config` must contain.

File: testbed/tests_unstable_exe/log_test.go [hunk 4]
```diff
@@ -31,4 +34,42 @@ func TestMain(m *testing.M) {
 	testbed.DoTestMain(m, contribPerfResultsSummary)
 }
 
-// This file is left with no tests as a placeholder for future unstable exe tests
+func TestLog10kDPS(t *testing.T) {
```
Comment (tigrannajaryan) [inline testbed/tests_unstable_exe/log_test.go:37]: Can we move this to the stable log_test.go instead?

Comment (sumo-drosiek) [reply testbed/tests_unstable_exe/log_test.go:37]: sure, I was not sure what exactly `unstable` means

Comment (tigrannajaryan) [reply testbed/tests_unstable_exe/log_test.go:37]: The `tests_unstable_exe/log_test.go` is for testing the unstable executable.
Stanza/filelog receiver previously was part of unstable executable. It was moved to `stable` exe yesterday.

**Comment by sumo-drosiek** [comment]
@tigrannajaryan All fixed, but Ci seems to be unstable Today

**Comment by tigrannajaryan** [comment]
I think there is a linting error (in addition to failing load test, which is probably CircleCI fault).

**Comment by sumo-drosiek** [comment]
@tigrannajaryan I doubt this is lint error related to this PR, as all recent PRs have same issue eg. https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector-contrib/10631/workflows/5e23abf9-f769-4d99-908e-c046ea55bbda/jobs/89573 

```
make[1]: Entering directory '/home/circleci/project'
golangci-lint run --allow-parallel-runners
ERRO Running error: context loading failed: no go files to analyze 
make[1]: *** [Makefile.Common:102: lint] Error 5
```

**Comment by sumo-drosiek** [comment]
@tigrannajaryan I added more tests in this PR and example is available in PR #2606

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan I doubt this lint error related to this PR, as all recent PRs have same issue eg. https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector-contrib/10631/workflows/5e23abf9-f769-4d99-908e-c046ea55bbda/jobs/89573
> 
> ```
> make[1]: Entering directory '/home/circleci/project'
> golangci-lint run --allow-parallel-runners
> ERRO Running error: context loading failed: no go files to analyze 
> make[1]: *** [Makefile.Common:102: lint] Error 5
> ```

Rebasing from main should help. It was broken.

File: testbed/datasenders/k8s.go [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+	config string
+}
+
+// Ensure FileLogK8sWriter implements LogDataSender.
+var _ testbed.LogDataSender = (*FileLogK8sWriter)(nil)
+
+// NewFileLogK8sWriter creates a new data sender that will write kubernetes containerd
+// log entries to a file, to be tailed by FileLogReceiver and sent to the collector.
+//
+// config is an OTC config appended to the receivers section after executing fmt.Sprintf on it.
```
Comment (tigrannajaryan) [inline testbed/datasenders/k8s.go:43]: What is OTC? If it refers to collector just use Collector or Otel Collector or Otelcol. We don't normally use OTC.

File: testbed/datasenders/k8s.go [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+// This impicated few things:
```
Comment (tigrannajaryan) [inline testbed/datasenders/k8s.go:44]:
```suggestion
// This implies a few things:
```

**Comment by djaglowski** [comment]
@sumo-drosiek The [latest release](https://github.com/open-telemetry/opentelemetry-log-collection/releases/tag/v0.16.0) of `opentelemetry-log-collection` will require some minor changes to this code. 

This should be a very straightforward change from using the term `labels` instead of `attributes`.

This won't be necessary until [this PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2607) is merged.

File: testbed/datasenders/k8s.go (hunk 5 above)
Comment (sumo-drosiek) [reply testbed/datasenders/k8s.go:43]: I will use `Otelcol` then

File: internal/stanza/register.go [hunk 7]
```diff
@@ -15,9 +15,12 @@
...
 import (
-	// Register parsers for stanza-based log receivers
+	// Register parsers and transformers for stanza-based log receivers
 	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/json"
 	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/regex"
 	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/severity"
 	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/time"
+	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/metadata"
+	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/restructure"
+	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/router"
```
Comment (sumo-drosiek) [inline internal/stanza/register.go:25]: I was unable to make linter happy with comment above `transformer` section. At least one of two linters has been always raising some issues:

```
Check License finished successfully
internal/stanza/register.go: Import groups are not in the proper order: ["Third party" "Third party"]
```
or
```
register.go:22: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)
        _ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/time"
```
or
```
register.go:23: File is not `gofmt`-ed with `-s` (gofmt)

```

I tried three different approaches
```
	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/time"
	// comment here
	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/metadata"
```

```
	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/time"

	// comment here
	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/metadata"
```

```
	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/parser/time"
        
	// comment here (preceding line is tab indented)
	_ "github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/metadata"
```

Comment (djaglowski) [reply internal/stanza/register.go:25]: No need to split into two groups - looks good to me like this.

**Review by djaglowski — APPROVED** [review]
LGTM

**Comment by sumo-drosiek** [comment]
@tigrannajaryan @djaglowski Everything addressed, CI is green :)

**Comment by tigrannajaryan** [comment]
Thank you @sumo-drosiek

**Pull request merged**